provider/datagen/src/baked_exporter.rs

```diff
@@ -334,32 +342,34 @@ impl DataExporter for BakedExporter {
             .replace('@', "_v")
             .replace('/', "_");
 
-        let mut singleton = None;
-        let mut singleton_ident = None;
+        let mut lookup_macro = None;
+        let mut lookup_ident = None;
+        let mut public_lookup_macro = false;
 
-        let lookup = match values.iter().map(|(_, l)| l.len()).sum() {
+        let lookup_ref = match values.iter().map(|(_, l)| l.len()).sum() {
             0 => quote!(None),
             1 => {
                 let (bake, locale) = values.into_iter().next().unwrap();
                 let locale = locale.into_iter().next().unwrap();
 
-                singleton_ident =
-                    Some(format!("singleton_{ident}").parse::<TokenStream>().unwrap());
-                let singleton_ident_prefixed = format!("__singleton_{ident}")
+                lookup_ident = Some(format!("singleton_{ident}").parse::<TokenStream>().unwrap());
+                let lookup_ident_prefixed = format!("__singleton_{ident}")
                     .parse::<TokenStream>()
                     .unwrap();
 
                 // Exposing singleton structs separately allows us to get rid of fallibility by using
                 // the struct directly.
-                singleton = Some(quote! {
+                lookup_macro = Some(quote! {
                     #[doc(hidden)]
                     #[macro_export]
-                    macro_rules! #singleton_ident_prefixed {
+                    macro_rules! #lookup_ident_prefixed {
                         () => {
                             #bake
                         }
                     }
                 });
+                // The singleton macros are public. Other lookup macros are not.
+                public_lookup_macro = true;
```

Comment:
chose not to reexport the non-singleton one since I can't see it being useful outside of a baked context.

Reply:
We need three macros for singletons: the singleton expression, the lookup expression, and the impl

Reply:
Ah you skip the lookup macro by duplicating the code, I guess that works but I'd prefer to always have a lookup macro